### PR TITLE
[receiver/awscontainerinsightsreceiver] Pod Detection changes for K8's in containerd runtime

### PR DESCRIPTION
**Pull request by vasireddy99**
**Description:**
[dockershim](https://kubernetes.io/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-removal-affects-you/) will be deprecated in the latest kubernetes release. Due to this following changes in PR are required when using `containerd` runtime

* k8s cluster using containerd runtime will not be able to get the pod metrics as containerd socket is not mounted
pod detection logic relies on pause container with name 'POD`, which is only the case for docker
 * Define a new constant TypeInfraContainer and remove most usage on POD container name.
  update all the manifest to mount containerd

EKS Announcement: [Amazon EKS is ending support for dockershim](https://docs.aws.amazon.com/eks/latest/userguide/dockershim-deprecation.html) 

**Link to tracking Issue:**

 * https://github.com/aws-observability/aws-otel-collector/issues/1071

**Documentation:**

* Read me changes to example deployment template is modified
* Logged in `CHANGELOG.md`

### CODE REVIEW

File: receiver/awscontainerinsightreceiver/README.md [hunk 1]
```diff
@@ -303,6 +306,9 @@ spec:
         - name: varlibdocker
```
[the remark this answers is not in the document]
Comment (vasireddy99) [reply receiver/awscontainerinsightreceiver/README.md:306]: it is safe to retain the path, [reference](https://github.com/aws-samples/amazon-cloudwatch-container-insights/blob/master/k8s-deployment-manifest-templates/deployment-mode/daemonset/combination/combination.yaml#L150-L162) from the container-d supported version of yaml. 
and in viceversa tested this from local on EKS 1.22 version with this yaml file 
image

File: receiver/awscontainerinsightreceiver/internal/cadvisor/container_info_processor.go [hunk 2]
```diff
@@ -117,7 +117,9 @@ func processContainer(info *cInfo.ContainerInfo, mInfo extractors.CPUMemInfoProv
 		namespace := info.Spec.Labels[namespaceLabel]
 		podName := info.Spec.Labels[podNameLabel]
 		podID := info.Spec.Labels[podIDLabel]
-		if containerName == "" || namespace == "" || podName == "" {
+		// NOTE: containerName can be empty for pause container on containerd
```
[the remark this answers is not in the document]
Comment (vasireddy99) [reply receiver/awscontainerinsightreceiver/internal/cadvisor/container_info_processor.go:120]: It looks like there actually isn't a way to do this validation. :confused:

File: unreleased/detect-pod.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```
Comment (Aneurysm9) [inline unreleased/detect-pod.yaml:2]: This isn't a breaking change, is it?  It doesn't change exported APIs at all.  It seems more like an enhancement to me.

Comment (vasireddy99) [reply unreleased/detect-pod.yaml]: Agreed, thanks for the suggestion

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**